pkg/kubelet/apis/kubeletconfig/v1beta1/defaults.go

```diff
@@ -49,11 +47,32 @@ func SetDefaults_KubeletConfiguration(obj *KubeletConfiguration) {
 	if obj.ConfigTrialDuration == nil {
 		obj.ConfigTrialDuration = &metav1.Duration{Duration: 10 * time.Minute}
 	}
+	if obj.SyncFrequency == zeroDuration {
+		obj.SyncFrequency = metav1.Duration{Duration: 1 * time.Minute}
+	}
+	if obj.FileCheckFrequency == zeroDuration {
+		obj.FileCheckFrequency = metav1.Duration{Duration: 20 * time.Second}
+	}
+	if obj.HTTPCheckFrequency == zeroDuration {
+		obj.HTTPCheckFrequency = metav1.Duration{Duration: 20 * time.Second}
+	}
+	if obj.EnableServer == nil {
+		obj.EnableServer = utilpointer.BoolPtr(true)
+	}
+	if obj.Address == "" {
+		obj.Address = "0.0.0.0"
+	}
+	if obj.Port == 0 {
+		obj.Port = ports.KubeletPort
+	}
+	if obj.ReadOnlyPort == nil {
+		obj.ReadOnlyPort = utilpointer.Int32Ptr(ports.KubeletReadOnlyPort)
+	}
 	if obj.Authentication.Anonymous.Enabled == nil {
-		obj.Authentication.Anonymous.Enabled = boolVar(true)
+		obj.Authentication.Anonymous.Enabled = utilpointer.BoolPtr(true)
```

Comment:
cc @tallclair 
is this our last chance to change kubelet defaults from being insecure by default (listen on 0.0.0.0 with no authn/authz)? promoting these defaults to beta makes me sad.

Reply:
I didn't think we could change any of the defaults in ways that could break CLI compatibility (--anonymous-auth in this case), but
IMO security > backwards compatibility, so I'm ok with this sort of change as long as there's a clear announcement. @tallclair WDYT?

Reply:
xref https://github.com/kubernetes/kubernetes/issues/53618

Reply:
I don't think we can break backwards compatibility of the flags, but here's an idea:
Since the eventual goal is to deprecate the flags, could we keep the alpha API, and keep the flags mapping to the alpha version? Then we can change the defaults used in beta without breaking backwards compatibility. This would mean some divergence, but I think the version conversion machinery should take care of most of that?

Reply:
I guess since we parse flags into the internal, we can continue to construct the initial internal from the v1alpha1 defaults while loading config files from the v1beta1 format. 

This means the v1alpha1 will need to stick around for several releases... which I don't really like. But I do think the chance at better defaults may be worth it.

Reply:
Could we instead just override v1beta1 defaults in `options.NewKubeletConfiguration` to preserve legacy values for the command line?